src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets

```diff
@@ -2,18 +2,48 @@
 
 <Project>
  
-  <UsingTask TaskName="ValidatePackage" AssemblyFile="$(DotNetPackageValidationAssembly)"/>
-  
+  <UsingTask TaskName="ValidatePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
+  <UsingTask TaskName="GetLastStablePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
+
   <Target Name="RunPackageValidation" AfterTargets="Pack">
+    
+    <GetLastStablePackage PackageId="$(PackageId)"
+      LocalPackagesPath="$(NuGetPackageRoot)$(PreviousPackageName)"
+      UseLocalPackagesPath="true" >
+      <Output TaskParameter="LastStableVersion" PropertyName="PreviousPackageVersion" />
+    </GetLastStablePackage>
+    
     <PropertyGroup>
       <PackagePath Condition="'$(PackagePath)' == ''">$(PackageOutputAbsolutePath)$(PackageId).$(PackageVersion).nupkg</PackagePath>
+      <PreviousPackagePath>$(NuGetPackageRoot)$(PreviousPackageName)\$(PreviousPackageVersion)\$(PreviousPackageName).$(PreviousPackageVersion).nupkg</PreviousPackagePath>
     </PropertyGroup>
 
     <ValidatePackage PackagePath="$(PackagePath)"
+                     PreviousPackagePath="$(PreviousPackagePath)"
                      RuntimeGraph="$(RuntimeGraph)"
                      NoWarn="$(NoWarn)"
                      RunApiCompat="$([MSBuild]::ValueOrDefault('$(RunApiCompat)', 'false'))"
                      ValidateWithPreviousVersion="$([MSBuild]::ValueOrDefault('$(ValidateWithPreviousVersion)', 'false'))"/>
   </Target>
 
+  <Target Name="AddPackageDownload" BeforeTargets="CollectPackageDownloads">
+
+    <ItemGroup>
+      <Feeds Include="$(_OutputSources)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <PreviousPackageName>$([MSBuild]::ValueOrDefault('$(PreviousPackageName)', '$(MSbuildProjectName)'))</PreviousPackageName>
```

Comment:
nit: casing of `MSBuildProjectName` is wrong.

Reply:
As this property is used in the other target as well, should it be defined centrally / moved into a target which the other two targets depend on?

Reply:
Double check the property used by NuGet for PackageId. https://github.com/NuGet/NuGet.Client/blob/f24bad0668193ce21a1db8cabd1ce95ba509c7f0/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L30